factory.go

```diff
@@ -28,21 +38,67 @@ func (f factory) DeploymentType() deployer.DeploymentType {
 	return "python"
 }
 
-func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*Config] {
+func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*config.Config] {
 	return Schema
 }
 
-func (f factory) Create(config *Config, logger log.Logger) (deployer.Connector, error) {
+func (f factory) NextConnectorIndex() int {
+	return int(atomic.AddInt64(f.connectorCounter, 1))
+}
+
+func (f factory) Create(config *config.Config, logger log.Logger) (deployer.Connector, error) {
 	pythonPath, err := binaryCheck(config.PythonPath)
 	if err != nil {
-		return &Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+		return &connector.Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+	}
+
+	pythonSemver := config.PythonSemVer
+	if pythonSemver == "" {
+		outputSemver, err := f.parsePythonVersion(pythonPath)
+		if err != nil {
+			return nil, err
+		}
+		pythonSemver = *outputSemver
+	}
+
+	connectorFilename := strings.Join([]string{
+		"connector",
+		strings.Replace(pythonSemver, ".", "-", -1),
+		strconv.Itoa(f.NextConnectorIndex())},
+		"_")
+
+	absWorkDir, err := filepath.Abs(config.WorkDir)
+	if err != nil {
+		return nil, fmt.Errorf("error determining absolute path for python deployer's working directory (%w)", err)
+	}
+	connectorFilepath := filepath.Join(absWorkDir, connectorFilename)
+	err = os.MkdirAll(connectorFilepath, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error creating temporary directory for python connector (%w)", err)
+	}
+
+	pythonCli := cliwrapper.NewCliWrapper(pythonPath, connectorFilepath, logger)
+
+	cn := connector.NewConnector(
+		config, logger, connectorFilepath, pythonCli)
+	return &cn, nil
+}
+
+// parsePythonVersion function gets the output of a command that asks the
+// Python executable for its semantic version string.
+func (f factory) parsePythonVersion(pythonPath string) (*string, error) {
+	versionCmd := exec.Command(pythonPath, "--version")
+	output, err := versionCmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	re, err := regexp.Compile(`\d+\.\d+\.\d+`)
+	if err != nil {
+		return nil, err
 	}
-	python := cliwrapper.NewCliWrapper(pythonPath, config.WorkDir, logger)
-	return &Connector{
-		config:           config,
-		logger:           logger,
-		pythonCliWrapper: python,
-	}, nil
+	found := re.FindString(string(output))
+	return &found, nil
```

Comment:
For this function, there wasn't a reason. I didn't think it mattered. I'll change it, but I don't think this will have a significant impact

Reply:
Ok, I think recall now. I think I would rather return `nil` for the string part of the return value when there is an error rather than an empty string.